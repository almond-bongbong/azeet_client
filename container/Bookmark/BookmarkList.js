import React from 'react';
import PropTypes from 'prop-types';
import BookmarkPreview from 'components/Bookmark/BookmarkPreview';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
import styled from 'styled-components';

const BookmarkWrapperStyle = styled.div`
  & + & {
    margin-top: 15px;
  }
  &.bookmark-enter {
    opacity: 0;
    transition: opacity .3s;
  }
  &.bookmark-enter-active {
    opacity: 1;
  }
  &.bookmark-leave {
    opacity: 1;
    transition: opacity .3s;
  }
  &.bookmark-leave-active {
    opacity: 0;
  }
`;

const BookmarkList = ({ bookmarks }) => (
  <div>
    <ReactCSSTransitionGroup
      transitionName="bookmark"
      transitionEnterTimeout={300}
      transitionLeaveTimeout={300}
    >
      {bookmarks.map(b => (
        <BookmarkWrapperStyle key={b.id}>
          <BookmarkPreview og={b.og} />
        </BookmarkWrapperStyle>
      ))}
    </ReactCSSTransitionGroup>
  </div>
);

BookmarkList.propTypes = {
  bookmarks: PropTypes.arrayOf(PropTypes.shape({})),
};

BookmarkList.defaultProps = {
  bookmarks: [],
};

export default BookmarkList;
