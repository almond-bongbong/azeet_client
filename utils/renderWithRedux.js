import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configStore from 'store/configure';
import Alert from 'components/Alert';
import Confirm from 'components/Confirm';
import Toast from 'components/Toast';

function renderWithRedux(ui, initialState) {
  const store = configStore(initialState);
  const utils = render(
    <Provider store={store}>
      {ui}
      <Alert />
      <Confirm />
      <Toast />
    </Provider>,
  );
  return {
    ...utils,
    store,
  };
}

export default renderWithRedux;
