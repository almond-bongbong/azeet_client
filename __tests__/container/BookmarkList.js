import React from 'react';
import renderWithRedux from 'utils/renderWithRedux';
import BookmarkList from 'container/Bookmark/BookmarkList';
// import axios from 'axios';
// import { initAxios } from 'config/configureAxios';
// import MockAdapter from 'axios-mock-adapter';

describe('BookmarkList', () => {
  // initAxios();
  // const mock = new MockAdapter(axios, { delayResponse: 200 }); // 200ms 가짜 딜레이 설정
  // mock.onGet('/bookmark').reply(200, {
  //   items: [
  //     { id: 1, og: { title: '제목', description: 'og_description1' } },
  //     { id: 2, og: { title: '제목', description: 'og_description2' } },
  //   ],
  // });

  it('matches snapshot', () => {
    const utils = renderWithRedux(<BookmarkList />);
    expect(utils.container).toMatchSnapshot();
  });

  it('get bookmark list', async () => {
    const { findAllByText } = renderWithRedux(<BookmarkList />, {
      bookmark: {
        bookmarks: [
          { id: 1, og: { title: '제목', description: 'og_description1' } },
          { id: 2, og: { title: '제목', description: 'og_description2' } },
        ],
      },
    });
    const bookmarks = await findAllByText('제목');
    expect(bookmarks).toHaveLength(2);
  });
});
