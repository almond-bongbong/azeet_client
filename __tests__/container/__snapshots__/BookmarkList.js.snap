// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarkList matches snapshot 1`] = `
<div>
  <div>
    <div
      class="SimpleLoader__SimpleLoaderStyle-sc-1c1yu7o-0 YjQN"
      size="60"
    >
      <div
        class="circle-loader"
        title="로딩중"
      />
    </div>
    <span />
  </div>
  <span />
  <span />
  <div
    class="Toast__ToastStyle-sc-18p3u8j-0 jhiKCx"
  >
    <span />
  </div>
</div>
`;
